components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -252,7 +252,10 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
 		if err != nil {
 			return admission.Errored(http.StatusInternalServerError, err)
 		}
+	}
 
+	// Check Imagestream Info both on create and update operations
+	if req.Operation == admissionv1.Create || req.Operation == admissionv1.Update {
```

Comment:
Check the approach, 
the admission.update would be triggered for any event on the notebook creation, which would trigger the feature every few secs.
irrespective of this, 
incase of running workbench, if the operator upgrade the notebook images, this would update a long running notebook directly. we need to protect this.

how about, instead of checking for admission webhook on update, we do 
check on if image of workbench was changed (this would catch the edit scenario) and also check if the workbench is stopped, then we check as well (this would take care of long running notebook with source image to be updated as well).

possible suggestion:
```suggestion
	stoppedAnnotation := notebook.ObjectMeta.Annotations[culler.STOP_ANNOTATION]
	_, isStopped := time.Parse(time.RFC3339, stoppedAnnotation)

	isImageChanged := strings.Contains(notebook.Spec.Template.Spec.Containers[0].Image, "image-registry.openshift-image-registry.svc:5000")

	if (!isImageChanged) && (req.Operation == admissionv1.Create || isStopped == nil) {
	
```